src/server/config/schema.js

```diff
@@ -52,6 +52,7 @@ module.exports = () => Joi.object({
       disableProtection: Joi.boolean().default(false),
       token: Joi.string().optional().notes('Deprecated')
     }).default(),
+    staticDirectoryListings: Joi.boolean().default(Joi.ref('$dev')),
```

Comment:
With this implementation, it seems impossible to disable the static directory listings in development.  Setting `server.staticDirectoryListings: false` doesn't seem to have any effect in development.

How about we just default this to `false` and then people can permanently set it in their kibana.dev.yml if they want to change that?